managed/services/management/service_test.go

```diff
@@ -256,4 +256,134 @@ func TestServiceService(t *testing.T) {
 			tests.AssertGRPCError(t, status.New(codes.NotFound, fmt.Sprintf(`Node with ID "%s" not found.`, node.NodeID)), err)
 		})
 	})
+
+	t.Run("AddCustomLabels", func(t *testing.T) {
+		t.Run("No Service ID", func(t *testing.T) {
+			ctx, s, teardown, _ := setup(t)
+			defer teardown(t)
+
+			response, err := s.AddCustomLabels(ctx, &managementpb.AddCustomLabelsRequest{})
+			assert.Nil(t, response)
+			tests.AssertGRPCError(t, status.New(codes.InvalidArgument, "service_id is required"), err)
+		})
+
+		t.Run("Add a label", func(t *testing.T) {
+			ctx, s, teardown, vmdb := setup(t)
+			defer teardown(t)
+
+			vmdb.Mock.On("RequestConfigurationUpdate").Once().Return()
```

Comment:
The `Once()` call makes sense together with
```
mock.AssertExpectationsForObjects(t, vmdb.Mock)
```
which checks that all mocked methods are actually being called and called the exact amount of times. Otherwise tests with mocks may end up with a lot of dead code and build an incorrect picture of how the function really works.

Reply:
I may be missing something, but this shall be checked during `teardown` in https://github.com/percona/pmm/blob/65291e6c78f03d8b9bbe436a64db3d436b6acc68/managed/services/management/service_test.go#L60

Is the approach you're suggesting different in some ways to the `teardown` approach?

Reply:
That's what I've missed. Then it's totally fine, I don't think there is any sufficient difference.